### ISSUE TITLE
Consensus limits for Private Sends

### ISSUE THREAD

**Issue by riordant** [zcoinofficial/zcoin-client#81]
The client currently does not block when spends requested exceeds the consensus limit per transaction or block. There are a couple of options here:
- allow the user to enter arbitrary amount, and allow the daemon to handle this case
- Send consensus limit data to the client in some form, and have the client enforce the transaction limit from the UI. This is not difficult  - the limit is a set value of spends OR value.
I personally prefer the second option. Thoughts @reubenyap @sproxet

**Comment by reubenyap** [comment]
Client should enforce transaction limit from UI.
Isn't limit both number of spends AND value?

**Comment by riordant** [comment]
It's OR:
https://github.com/zcoinofficial/zcoin/blob/master/src/miner.cpp#L421
If either condition is there it blocks

**Comment by riordant** [comment]
I think we will go with that option then.
Daemon: send consensus data. This data should only have to be sent once at startup. Could do via `apiStatus`, but seems a bit of a waste as it only needs to be sent once. I could add a `consensus` REPlier method with eg.
```
consensus: {
    perBlock: {
        inputs: x
        value: x
    }
    perTx: {
        inputs: x
        value:x
    }
}
```
What do you think @sproxet

**Comment by sproxet** [comment]
The client has no idea what block the transaction will be included in, so it can't obey block limits. And won't those just stay in the mempool until they get included in a block anyway? 8d8b2ede39 implemented validation that individual spends did not exceed 500 XZC (though that value is hardcoded). Validating number of inputs would require either the daemon to calculate it or for the daemon to push a list of unspent mints.
